### PR TITLE
added sysrepo initalizer to main(), update makefile, update gh actions

### PR DESCRIPTION
**Pull request by Ali-aqrabawi**
added initial sysrepo integration to iproute2_sysrepo's `main()`

### CODE REVIEW

File: .github/workflows/actions-compile.yml [hunk 1]
```diff
@@ -18,16 +18,32 @@ jobs:
...
         uses: actions/checkout@v4
         with:
           submodules: true
+      - name: libyang v2.1.148 from upstream
+        uses: actions/checkout@v4
+        with:
+            repository: 'CESNET/libyang.git'
+            ref: 'v2.1.148'
+      - name: make libyang v2.1.148 from upstream
+        run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE:String="Release" .. && make -j $(nproc) && make install
```
Comment (vjardin) [inline .github/workflows/actions-compile.yml:32]: && sudo make install ?

File: .github/workflows/actions-compile.yml [hunk 2]
```diff
@@ -18,16 +18,32 @@ jobs:
...
+      - name: sysrepo v2.2.105 from upstream
+        uses: actions/checkout@v4
+        with:
+          repository: 'sysrepo/sysrepo.git'
+          ref: 'v2.2.105'
+      - name: make sysrepo v2.2.150 from upstream
+        run: mkdir build && cd build && cmake -DCMAKE_INSTALL_PREFIX:PATH=$(pwd)/bin -DCMAKE_BUILD_TYPE:String="Release" .. && make -j $(nproc) && make install
```
Comment (vjardin) [inline .github/workflows/actions-compile.yml:39]: what's about removing the path and doing a sudo make install ?

**Review by vjardin — CHANGES_REQUESTED** [review]

File: .github/workflows/actions-compile.yml (hunk 2 above)
Comment (Ali-aqrabawi) [reply .github/workflows/actions-compile.yml:39]: i'm trying to run the action locally, once done i will try that,

File: .github/workflows/actions-compile.yml [hunk 3]
```diff
@@ -18,16 +18,34 @@ jobs:
       - name: add missing packages
         uses: ConorMacBride/install-package@v1
         with:
-          apt: libelf-dev libbpf-dev libbsd-dev libsysrepo-dev libcap-dev libmnl-dev libselinux-dev libyang-tools
+          apt: libelf-dev libbpf-dev libbsd-dev libcap-dev libmnl-dev libselinux-dev cmake make gcc libtool-bin
```
Comment (vjardin) [inline .github/workflows/actions-compile.yml:21]: why adding make and gcc, there are already installed:
  https://github.com/okda-networks/iproute2-sysrepo/actions/runs/7967215693/job/21749657929#step:2:47

File: .github/workflows/actions-compile.yml [hunk 4]
```diff
@@ -18,16 +18,34 @@ jobs:
...
+        run: sudo ldconfig
+      - name: check sysrepo running ok
+        run: sudo ldconfig && sudo sysrepoctl -l
       - name: recursive checkout (with iproute2 submodule)
         uses: actions/checkout@v4
         with:
           submodules: true
       - name: make iproute2-sysrepo
-        run : CC=${{ matrix.compiler }} make -j $(nproc)
+        run:  CC=gcc make -j $(nproc)
```
Comment (vjardin) [inline .github/workflows/actions-compile.yml:45]: why do you drop clang ?

File: .github/workflows/actions-compile.yml [hunk 5]
```diff
@@ -18,16 +18,34 @@ jobs:
...
       - name: basic compilation check
-        run : ldd bin/iproute2-sysrepo
+        run: ldd bin/iproute2-sysrepo
       - name: basic test ip address like
-        run : bin/iproute2-sysrepo link
+        run: bin/iproute2-sysrepo link
       - name: yanglint
-        run : make check
+        run: make check
```
Comment (vjardin) [inline .github/workflows/actions-compile.yml:51]: missing the EOF char

File: .github/workflows/actions-compile.yml [hunk 6]
```diff
@@ -18,16 +18,34 @@ jobs:
...
-        run : CC=${{ matrix.compiler }} make -j $(nproc)
+        run:  CC=gcc make -j $(nproc)
       - name: basic compilation check
-        run : ldd bin/iproute2-sysrepo
+        run: ldd bin/iproute2-sysrepo
       - name: basic test ip address like
-        run : bin/iproute2-sysrepo link
+        run: bin/iproute2-sysrepo link
       - name: yanglint
-        run : make check
```
Comment (vjardin) [inline .github/workflows/actions-compile.yml:33]: Please, do not change the white spaces if not needed it unfocuses the code review.

**Review by vjardin — CHANGES_REQUESTED** [review]

File: Makefile [hunk 7]
```diff
@@ -45,3 +52,9 @@ iproute2/config.mk:
 
 check:
 	yanglint yang/*.yang
+
+$(IPR2_SR_LIB_OBJ): $(IPR2_SR_LIB_SRC)
+	$(CC) -c $< -o $@ -Iiproute2/ip -Iiproute2/include
+
+$(IPR2_SR_OBJ): $(IPR2_SR_SRC)
+	$(CC) -c $< -o $@ -Iiproute2/ip -Iiproute2/include
```
Comment (vjardin) [inline Makefile:60]: missing the EOF char

**Comment by vjardin** [comment]
@Ali-aqrabawi , moreover, can you squash some of the commits that were local trials of github's CI ?

**Pull request merged**